src/Microsoft.DotNet.Darc/DarcLib.AzDev/AzureDevOpsClient.cs

```diff
@@ -1179,19 +1179,35 @@ public async Task<int> StartNewReleaseAsync(string accountName, string projectNa
     /// <param name="azdoDefinitionId">ID of the build definition where a build should be queued.</param>
     /// <param name="queueTimeVariables">Queue time variables as a Dictionary of (variable name, value).</param>
     /// <param name="templateParameters">Template parameters as a Dictionary of (variable name, value).</param>
+    /// /// <param name="pipelineResources">Pipeline resources as a Dictionary of (pipeline resource name, build number).</param>
     public async Task<int> StartNewBuildAsync(
         string accountName,
         string projectName,
         int azdoDefinitionId,
         string sourceBranch,
         string sourceVersion,
         Dictionary<string, string> queueTimeVariables = null,
-        Dictionary<string, string> templateParameters = null)
+        Dictionary<string, string> templateParameters = null,
+        Dictionary<string, string> pipelineResources = null)
     {
         var variables = new Dictionary<string, AzureDevOpsVariable>();
-        foreach ((string name, string value) in queueTimeVariables)
+
+        if (queueTimeVariables != null)
         {
-            variables.Add(name, new AzureDevOpsVariable(value));
+            foreach ((string name, string value) in queueTimeVariables)
+            {
+                variables.Add(name, new AzureDevOpsVariable(value));
+            }
+        }
```

Comment:
Not sure if you'd prefer that but you could do:

```csharp
var variables = queueTimeVariables?.ToDictionary<>(...) ?? new Dictionary<string, AzureDevOpsVariable>();
```

Same below

Reply:
but queueTimeVariables is a dictionary - does it work like that?

Reply:
It should, you're basically doing the same thing with the `foreach`

Reply:
`Dictionary is also `IEnumerable<KeyValuePair<K, V>>`

Reply:
took some time but I figured it out